apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -88,57 +100,84 @@ const initialState: AichatState = {
 // output a message to the chat window with the list of customizations that were updated.
 export const updateAiCustomization = createAsyncThunk(
   'aichat/updateAiCustomization',
+  async (_, thunkAPI) => {
+    await updateAiCustomizationShared(thunkAPI);
+  }
+);
+
+export const publishModel = createAsyncThunk(
+  'aichat/publishModelCard',
+  async (_, thunkAPI) => {
+    thunkAPI.dispatch(setHasPublished(true));
+    await updateAiCustomizationShared(thunkAPI);
+    thunkAPI.dispatch(setViewMode(ViewMode.PRESENTATION));
+  }
+);
+
+export const saveModelCard = createAsyncThunk(
+  'aichat/saveModelCard',
   async (_, thunkAPI) => {
     const state = thunkAPI.getState() as RootState;
-    const {currentAiCustomizations, previouslySavedAiCustomizations} =
-      state.aichat;
+    const {modelCardInfo} = state.aichat.currentAiCustomizations;
 
-    // Remove any empty example topics on save
-    const trimmedExampleTopics =
-      currentAiCustomizations.modelCardInfo.exampleTopics.filter(
-        topic => topic.length
-      );
-    thunkAPI.dispatch(
-      setModelCardProperty({
-        property: 'exampleTopics',
-        value: trimmedExampleTopics,
-      })
+    if (!hasFilledOutModelCard(modelCardInfo)) {
+      thunkAPI.dispatch(setHasPublished(false));
+    }
+    await updateAiCustomizationShared(thunkAPI);
+  }
+);
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const updateAiCustomizationShared = async (thunkAPI: any) => {
```

Comment:
nit: maybe rename to `updateAndSaveAiCustomizations` or just `saveAiCustomizations`? Also I think if you just pass in `state` and `dispatch` as separate args, you should be able to get rid of the `any`? Maybe even reduce down to just the specific state fields you need (current/previousAiCustomizations). We have a similar pattern in Lab2 where different async thunks make use of common shared functions, and just pass in the specific data they need from the `thunkApi`. Ex: https://github.com/code-dot-org/code-dot-org/blob/f0320b0cffd1f042aefcabb1d8497ca0b4a277ce/apps/src/lab2/lab2Redux.ts#L388-L396

Reply:
>nit: maybe rename to updateAndSaveAiCustomizations or just saveAiCustomizations?

+1 on `saveAiCustomizations`

Reply:
Updated here -- went with `saveAiCustomization` for the thunk that is exported, and `saveAiCustomizationShared` for the shared code since it aligns exactly with what happens when you use `saveAiCustomization`. LMK if that works for y'all!

https://github.com/code-dot-org/code-dot-org/pull/57949/commits/a180faa64237d3628ae74d1752deedb097da0c9c

Reply:
It seems like `saveAiCustomizationShared` is similar to an internal or private function. Is there a convention we could establish for these 'shared' functions that are used internally? 'Shared 'seems to have multiple meanings so may be a little unclear to a reader.